lib/improver/wind_direction.py

```diff
@@ -0,0 +1,409 @@
+                                                    x_coord_name]):
+            # Extract wind direction data.
+            wind_dir_deg = slice_ens_wdir.data
+            realization_axis = slice_ens_wdir.coord_dims('realization')[0]
+
+            # Copies input cube and remove realization dimension to create
+            # cubes for storing results.
+            slice_mean_wdir = next(slice_ens_wdir.slices_over("realization"))
+            slice_mean_wdir.remove_coord('realization')
+
+            # Convert wind direction from degrees to complex numbers.
+            wind_dir_complex = WindDirection.deg_to_complex(wind_dir_deg)
+
+            # Find the complex average -  which actually signifies a point
+            # between all of the data points in POLAR co-cordinates.
+            # NOT the average DEGREE ANGLE.
+            wind_dir_complex_mean = np.mean(wind_dir_complex,
+                                            axis=realization_axis)
+
+            # Convert complex average values to degrees to produce average
+            # wind direction.
+            wind_dir_deg_mean = WindDirection.complex_to_deg(
+                wind_dir_complex_mean)
+
+            # Find radius values for wind direction average.
+            r_vals = WindDirection.find_r_values(wind_dir_complex_mean)
+
+            # Calculate standard deviation from polar mean.
+            # TODO: This will still need some further investigation.
+            #        This is will be the subject of another ticket.
+            polar_mean_std_dev = WindDirection.calc_polar_mean_std_dev(
+                wind_dir_complex, wind_dir_deg_mean, r_vals, r_thresh,
+                realization_axis)
+
+            # Finds any meaningless averages and subistuite with
+            # the wind direction taken from the first ensemble member.
+            wind_dir_deg_mean = WindDirection.wind_dir_decider(
+                wind_dir_deg, wind_dir_deg_mean, r_vals, r_thresh)
+
+            # Save data into cubes (create new cubes for r and std dev data).
+            slice_mean_wdir.data = wind_dir_deg_mean
+            slice_r_vals = slice_mean_wdir.copy(data=r_vals)
+            slice_std_dev = slice_mean_wdir.copy(data=polar_mean_std_dev)
+            # Append to cubelists.
+            wdir_cube_list.append(slice_mean_wdir)
+            r_vals_cube_list.append(slice_r_vals)
+            std_dev_cube_list.append(slice_std_dev)
+
+        # Combine cubelists into cube.
+        cube_mean_wdir = wdir_cube_list.merge_cube()
+        cube_r_vals = r_vals_cube_list.merge_cube()
+        cube_polar_mean_std_dev = std_dev_cube_list.merge_cube()
+
+        # Change cube identifiers.
+        cube_mean_wdir.add_cell_method(CellMethod("mean",
+                                                  coords="realization"))
+        cube_r_vals.long_name = "Avg wind dir r-vals"
+        cube_r_vals.units = None
+        cube_polar_mean_std_dev.long_name = "Avg wind dir std dev"
```

Comment:
Please can you change the name of this cube from cube_polar_mean_std_dev to something more understandable and the long name to some equivalent e.g. "wind direction confidence measure"